### PR TITLE
Deprecate various functions

### PR DESCRIPTION
**Pull request by bpbond**
@wlandau Lots of merge conflicts when I updated `upstream/master` because of file rearrangements in 8fd7f75d. Opening this just to cleanly check no problems.

Also a question: there are many references to `built()` in the documentation. Should they be removed, or changed to something else?

# Summary

Deprecate a number of functions.

# Related GitHub issues and pull requests

- Ref: #644 

# Checklist

- [X] I have read `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CONDUCT.md), and I agree to follow its rules.
- [x] I have listed any substantial changes in the [development news](https://github.com/ropensci/drake/blob/master/NEWS.md).
- [X] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/ropensci/drake/tree/master/tests/testthat) to confirm that any new features or functionality work correctly.
- [x] I have tested this pull request locally with `devtools::check()`
- [x] This pull request is ready for review.
- [x] I think this pull request is ready to merge.

### CODE REVIEW

**Comment by wlandau** [comment]
> @wlandau Lots of merge conflicts when I updated upstream/master because of file rearrangements in 8fd7f75. Opening this just to cleanly check no problems.

Yeah, sorry about the merge conflicts. Thanks for doing this!

> Also a question: there are many references to built() in the documentation. Should they be removed, or changed to something else?

I think we should point them to `cached()`.

File: R/utils-deprecate.R [hunk 1]
```diff
@@ -891,6 +891,83 @@ drake_strings <- function(...) {
   out
 }
 
+#' @title Deprecated. List all the built targets (non-imports) in the cache.
+#' @description Targets are listed in the workflow plan
```
Comment (wlandau) [inline R/utils-deprecate.R]: Would you include the yyyy-mm-dd date that `built()` was deprecated? Same goes for the others.

Comment (bpbond) [reply R/utils-deprecate.R]: 👍

File: R/utils-deprecate.R [hunk 2]
```diff
@@ -891,6 +891,85 @@ drake_strings <- function(...) {
...
+built <- function(
+  path = getwd(), search = TRUE,
+  cache = drake::get_cache(path = path, search = search, verbose = verbose),
+  verbose = 1L,
+  jobs = 1
+) {
+  .Deprecated(
+    "built",
+    package = "drake",
+    msg = paste("built() is deprecated.")
```
Comment (wlandau) [inline R/utils-deprecate.R]: One last thing: maybe suggest `cached(no_imported_objects = TRUE))`?

**Comment by wlandau** [comment]
Fantastic, @bpbond! Great work.

**Pull request merged**